src/transaction.rs

```diff
@@ -15,6 +15,17 @@ pub const SIGNED_DATA_OFFSET: usize = size_of::<Signature>();
 pub const SIG_OFFSET: usize = 0;
 pub const PUB_KEY_OFFSET: usize = size_of::<Signature>() + size_of::<u64>();
 
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Clone)]
+pub struct Program {
```

Comment:
If you rebase, you can rename this to `Instruction` now.  The term `Program` is awkward here because that `program_id` field is not a unique identifier for instances of this struct, and because the stuff serialized in the `userdata` may be a new program or a call into an existing one.  Maybe this should be an enum too?